src/app/store/transactions/transaction.hooks.ts

```diff
@@ -125,6 +128,8 @@ export function useStacksTransactionBroadcast() {
       // Otherwise, it's a send form tx and we don't want to
       if (requestToken && tabId) {
         finalizeTxSignature({ requestPayload: requestToken, tabId, data: result });
+        await incrementPersistedNonce();
```

Comment:
Incrementing/decrementing the nonce from some pre-known seems like a more error-prone approach than saving what nonce was used for the latest successful transaction.

Could this instead be something to the effect of

```
await setLastKnownGoodNonceForAddress(address, nonce);
```